File: extension/js/LoggedPrice.js
// This data class represents a logged price for an item
class LoggedPrice{
    constructor(logDate, price, primePrice) {
        this.logDate = logDate;
        this.price = price;
        this.primePrice = primePrice;
    }

    priceFor(isPrime) {
        if (isPrime == true) {
            return this.primePrice
        } else {
            return this.price
        }
    } 

}

File: server/shared/logged_price.py
class LoggedPrice:
    def __init__(self, date, price, primePrice):
        self.date = date
        self.price = price
        self.primePrice = primePrice

    def __eq__(self, other):
        if isinstance(other, LoggedPrice):
            return (self.date == other.date 
                and self.primePrice == other.primePrice 
                and self.price == other.price)
        return False
